File: semgrep-rules/java/misc/testdata/SensitiveDataInLog.java
package com.example.testdata;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class SensitiveDataInLog {

    private static final Logger logger = LoggerFactory.getLogger(SensitiveDataInLog.class);

    // ruleid: java-misc-sensitive-data-in-log
    public void logPassword(String password) {
        logger.info("User login with password: {}", password);
    }

    // ruleid: java-misc-sensitive-data-in-log
    public void logToken(String accessToken) {
        logger.debug("Token issued: {}", accessToken);
    }

    // ruleid: java-misc-sensitive-data-in-log
    public void logSecretConcat(String secretKey) {
        logger.error("Key = " + secretKey);
    }

    // ruleid: java-misc-sensitive-data-in-log
    public void logApiKey(String apiKey) {
        logger.warn("API key received: {}", apiKey);
    }

    // ok: java-misc-sensitive-data-in-log
    public void logNormal(String username) {
        logger.info("User logged in: {}", username);
    }

    // ok: java-misc-sensitive-data-in-log
    public void logMessage() {
        logger.info("Please enter your password to continue");
    }

    // ok: java-misc-sensitive-data-in-log
    public void logUserId(int userId) {
        logger.debug("Processing request for user: {}", userId);
    }
}

File: semgrep-rules/javascript/misc/testdata/sensitive_data_in_log.js
// ruleid: javascript-misc-sensitive-data-in-log
function logPassword(password) {
    console.log("User password:", password);
}

// ruleid: javascript-misc-sensitive-data-in-log
function logToken(accessToken) {
    console.debug("Token: " + accessToken);
}

// ruleid: javascript-misc-sensitive-data-in-log
function logSecret(secretKey) {
    console.error(`Key = ${secretKey}`);
}

// ruleid: javascript-misc-sensitive-data-in-log
function logApiKey(apiKey) {
    console.warn("API key:", apiKey);
}

// ok: javascript-misc-sensitive-data-in-log
function logNormal(username) {
    console.log("User logged in:", username);
}

// ok: javascript-misc-sensitive-data-in-log
function logMessage() {
    console.log("Please enter your password");
}

// ok: javascript-misc-sensitive-data-in-log
function logUserId(userId) {
    console.debug("Processing user:", userId);
}
